### PR TITLE
Credrank Migration Full Launch

### PR DESCRIPTION
**Pull request by blueridger**
# Description
This is the full launch of the Credrank migration, as detailed by https://github.com/sourcecred/sourcecred/pull/2631/files
This involves three major changes:

1. Change the `go` command to no longer include `score`
1. Change the `grain` command to use credGraph instead of credView
1. Change the UI to no longer load CredView / the old explorer, and instead load CredGraph / the new explorer.

I also baked in a small change to add messaging to the new explorer in the case that the credGraph doesn't exist.

# Test Plan
## UI
1. scdev serve
1. Verify new Explorer Home loads
1. delete credGraph.json
1. hard refresh
1. Verify account page loads
1. Verify messaging:
Screen Shot 2021-03-15 at 2 43 48 PM

## Go command
1. checkout template-instance repo
1. add sourcecred/discourse to bundledPlugins
1. 

```
thena@thenas-mbp template-instance % scdev go
  GO   load
  GO   loading sourcecred/discourse
  GO   sourcecred/discourse: discourse
  GO   sourcecred/discourse: discourse/topics
 DONE  sourcecred/discourse: discourse/topics: 54s
  GO   sourcecred/discourse: discourse/likes
 DONE  sourcecred/discourse: discourse/likes: 5414ms
 DONE  sourcecred/discourse: discourse: 59s
 DONE  loading sourcecred/discourse: 59s
 DONE  load: 59s
  GO   graph
  GO   reference detector
  GO   reference detector for sourcecred/discourse
 DONE  reference detector for sourcecred/discourse: 2ms
 DONE  reference detector: 4ms
  GO   sourcecred/discourse: generating graph
 DONE  sourcecred/discourse: generating graph: 39ms
  GO   sourcecred/discourse: writing graph
 DONE  sourcecred/discourse: writing graph: 10ms
 DONE  graph: 59ms
  GO   credrank
  GO   load data
 DONE  load data: 7ms
  GO   run CredRank
 DONE  run CredRank: 94ms
# Top Participants By Cred

| Description | Cred | % |
| --- | --- | --- |
| d11 | 0.4 | 90.1% |
| dl-proto | 0.0 | 9.9% |
| beanow-sc-test | 0.0 | 0.0% |
| system | 0.0 | 0.0% |
| SourceCred | 0.0 | 0.0% |
  GO   writing changes
 DONE  writing changes: 5ms
 DONE  credrank: 111ms
```

## Grain command
1. checkout gh-pages branch of cred repo
1. manually delete last grain distribution from ledger
1. delete credResult.json
1. 

```
thena@thenas-mbp cred % scdev grain
 Distributed 25,000g to 81 identities in 1 distributions 

## GRAIN Distribution
### 2021-03-14T00:00:00.000Z
#### Total Distributed: 25,000g
#### RECENT: 20,000g
#### BALANCED: 5,000g
|          name          |    total    |     %     |
| ---------------------- | ----------- | --------- |
|         Thena          |  3,389.201  |  13.56%   |
|      panchomiguel      |  2,252.757  |   9.01%   |
|        KuraFire        |  1,846.743  |   7.39%   |
|       topocount        |  1,792.936  |   7.17%   |
|          Jojo          |  1,715.302  |   6.86%   |
|       echojuliet       |  1,613.237  |   6.45%   |
|       lbstrobbe        |  1,326.257  |   5.31%   |
|      joiecousins       |  1,115.598  |   4.46%   |
|         hammad         |  1,009.163  |   4.04%   |
|      decentralion      |   967.796   |   3.87%   |
|         s-ben          |   794.880   |   3.18%   |
|       sandpiper        |   749.046   |   3.00%   |
|          bex           |   698.686   |   2.79%   |
|      saintmedusa       |   647.855   |   2.59%   |
|     BrianBelhumeur     |   624.147   |   2.50%   |
|        wchargin        |   458.214   |   1.83%   |
|        magwalk         |   416.634   |   1.67%   |
|        DeepWork        |   366.250   |   1.47%   |
|          eeli          |   308.107   |   1.23%   |
|         Willow         |   282.008   |   1.13%   |
|         Felix          |   237.429   |   0.95%   |
|          tua           |   234.537   |   0.94%   |
|        benoxmo         |   226.851   |   0.91%   |
|         Marcie         |   211.039   |   0.84%   |
|       EbonyBelle       |   165.767   |   0.66%   |
|       ALOYSIOUS        |   164.726   |   0.66%   |
|         Kokomo         |   133.738   |   0.53%   |
|    youngkidwarrior     |   133.357   |   0.53%   |
|          Zizi          |   119.423   |   0.48%   |
|          ian           |   108.437   |   0.43%   |
|          peth          |   101.874   |   0.41%   |
|      coopahtroopa      |   97.140    |   0.39%   |
|         crisog         |   97.084    |   0.39%   |
|      TheCryptoCPA      |   91.653    |   0.37%   |
|         Threm          |   74.356    |   0.30%   |
|        prose11         |   60.991    |   0.24%   |
|        cortanav        |   54.082    |   0.22%   |
|         beanow         |   53.759    |   0.22%   |
|         amico          |   39.552    |   0.16%   |
|          evan          |   38.560    |   0.15%   |
|         yalor          |   38.437    |   0.15%   |
|          kus           |   27.503    |   0.11%   |
|        yetifree        |   16.566    |   0.07%   |
|         alive          |   16.438    |   0.07%   |
|         Dazuck         |   13.234    |   0.05%   |
|      monstrosity       |   12.772    |   0.05%   |
|        mozilla         |   10.331    |   0.04%   |
|         diego          |    6.004    |   0.02%   |
|       Jzjallday        |    5.770    |   0.02%   |
|      OttoZastrow       |    5.761    |   0.02%   |
|       metaverde        |    4.798    |   0.02%   |
|        mzargham        |    4.385    |   0.02%   |
|        cryptouf        |    3.573    |   0.01%   |
|   Vanilladelphia-eth   |    3.444    |   0.01%   |
|       johnny1093       |    2.892    |   0.01%   |
|       deltafreq        |    1.967    |   0.01%   |
|       stargarden       |    1.778    |   0.01%   |
|         sriche         |    1.610    |   0.01%   |
|        santigs         |    1.208    |   0.00%   |
|       Most-Evil        |    0.919    |   0.00%   |
|     doctorrobinson     |    0.460    |   0.00%   |
|         EthMan         |    0.403    |   0.00%   |
|      brianlitwin       |    0.170    |   0.00%   |
|        jimpick         |    0.144    |   0.00%   |
|        burrrata        |    0.094    |   0.00%   |
|          rpp           |    0.079    |   0.00%   |
|       ecwireless       |    0.036    |   0.00%   |
|     flyingzumwalt      |    0.013    |   0.00%   |
|         jbenet         |    0.003    |   0.00%   |
|        protocol        |    0.000    |   0.00%   |
|        zixuanzh        |    0.000    |   0.00%   |
|         noman          |    0.000    |   0.00%   |
|        pet3rpan        |    0.000    |   0.00%   |
|       sourcecred       |    0.000    |   0.00%   |
|        bpierre         |    0.000    |   0.00%   |
|         autark         |    0.000    |   0.00%   |
|        Carla78         |    0.000    |   0.00%   |
|        Macgyver        |    0.000    |   0.00%   |
|        marcvi1a        |    0.000    |   0.00%   |
|      beltenebros       |    0.000    |   0.00%   |
|        gilgahex        |    0.000    |   0.00%   |
RECENT Policy
Budget 20,000.000g
Discount: 0.16
|          name          |    total    |     %     |
| ---------------------- | ----------- | --------- |
|         Thena          |  2,794.699  |  13.97%   |
|      panchomiguel      |  1,774.084  |   8.87%   |
|       topocount        |  1,521.666  |   7.61%   |
|          Jojo          |  1,436.575  |   7.18%   |
|       echojuliet       |  1,366.315  |   6.83%   |
|        KuraFire        |  1,307.465  |   6.54%   |
|       lbstrobbe        |   996.715   |   4.98%   |
|      decentralion      |   967.796   |   4.84%   |
|         hammad         |   830.632   |   4.15%   |
|      joiecousins       |   727.066   |   3.64%   |
|         s-ben          |   693.239   |   3.47%   |
|      saintmedusa       |   549.779   |   2.75%   |
|     BrianBelhumeur     |   533.972   |   2.67%   |
|          bex           |   468.697   |   2.34%   |
|       sandpiper        |   463.874   |   2.32%   |
|        wchargin        |   458.214   |   2.29%   |
|        magwalk         |   354.819   |   1.77%   |
|        DeepWork        |   286.187   |   1.43%   |
|          eeli          |   243.305   |   1.22%   |
|         Willow         |   236.671   |   1.18%   |
|         Felix          |   199.545   |   1.00%   |
|          tua           |   189.053   |   0.95%   |
|         Marcie         |   168.238   |   0.84%   |
|       ALOYSIOUS        |   143.341   |   0.72%   |
|        benoxmo         |   133.499   |   0.67%   |
|       EbonyBelle       |   128.717   |   0.64%   |
|    youngkidwarrior     |   117.113   |   0.59%   |
|         Kokomo         |   107.585   |   0.54%   |
|          Zizi          |   103.515   |   0.52%   |
|          peth          |   82.794    |   0.41%   |
|      TheCryptoCPA      |   77.714    |   0.39%   |
|      coopahtroopa      |   67.294    |   0.34%   |
|         Threm          |   61.817    |   0.31%   |
|         beanow         |   53.759    |   0.27%   |
|         crisog         |   53.326    |   0.27%   |
|        prose11         |   50.514    |   0.25%   |
|         amico          |   39.552    |   0.20%   |
|          evan          |   38.560    |   0.19%   |
|         yalor          |   38.437    |   0.19%   |
|        cortanav        |   21.008    |   0.11%   |
|          kus           |   17.782    |   0.09%   |
|          ian           |   14.064    |   0.07%   |
|        yetifree        |   13.274    |   0.07%   |
|         Dazuck         |   12.169    |   0.06%   |
|      monstrosity       |   11.900    |   0.06%   |
|       Jzjallday        |    5.770    |   0.03%   |
|         diego          |    5.347    |   0.03%   |
|        mzargham        |    4.385    |   0.02%   |
|       metaverde        |    4.007    |   0.02%   |
|      OttoZastrow       |    3.732    |   0.02%   |
|        cryptouf        |    3.573    |   0.02%   |
|         alive          |    3.162    |   0.02%   |
|       johnny1093       |    2.466    |   0.01%   |
|   Vanilladelphia-eth   |    2.185    |   0.01%   |
|       deltafreq        |    1.967    |   0.01%   |
|         sriche         |    1.610    |   0.01%   |
|       stargarden       |    1.464    |   0.01%   |
|        santigs         |    1.208    |   0.01%   |
|       Most-Evil        |    0.919    |   0.00%   |
|     doctorrobinson     |    0.460    |   0.00%   |
|        mozilla         |    0.431    |   0.00%   |
|         EthMan         |    0.403    |   0.00%   |
|      brianlitwin       |    0.170    |   0.00%   |
|        jimpick         |    0.144    |   0.00%   |
|        burrrata        |    0.094    |   0.00%   |
|          rpp           |    0.079    |   0.00%   |
|       ecwireless       |    0.036    |   0.00%   |
|     flyingzumwalt      |    0.013    |   0.00%   |
|         jbenet         |    0.003    |   0.00%   |
|        protocol        |    0.000    |   0.00%   |
|        zixuanzh        |    0.000    |   0.00%   |
|         noman          |    0.000    |   0.00%   |
|        pet3rpan        |    0.000    |   0.00%   |
|       sourcecred       |    0.000    |   0.00%   |
|        bpierre         |    0.000    |   0.00%   |
|         autark         |    0.000    |   0.00%   |
|        Carla78         |    0.000    |   0.00%   |
|        Macgyver        |    0.000    |   0.00%   |
|        marcvi1a        |    0.000    |   0.00%   |
|      beltenebros       |    0.000    |   0.00%   |
|        gilgahex        |    0.000    |   0.00%   |
BALANCED Policy
Budget 5,000.000g
|          name          |    total    |     %     |
| ---------------------- | ----------- | --------- |
|         Thena          |   594.501   |  11.89%   |
|        KuraFire        |   539.278   |  10.79%   |
|      panchomiguel      |   478.672   |   9.57%   |
|      joiecousins       |   388.532   |   7.77%   |
|       lbstrobbe        |   329.542   |   6.59%   |
|       sandpiper        |   285.172   |   5.70%   |
|          Jojo          |   278.726   |   5.57%   |
|       topocount        |   271.270   |   5.43%   |
|       echojuliet       |   246.921   |   4.94%   |
|          bex           |   229.989   |   4.60%   |
|         hammad         |   178.531   |   3.57%   |
|         s-ben          |   101.641   |   2.03%   |
|      saintmedusa       |   98.076    |   1.96%   |
|          ian           |   94.373    |   1.89%   |
|        benoxmo         |   93.351    |   1.87%   |
|     BrianBelhumeur     |   90.175    |   1.80%   |
|        DeepWork        |   80.062    |   1.60%   |
|          eeli          |   64.801    |   1.30%   |
|        magwalk         |   61.814    |   1.24%   |
|          tua           |   45.484    |   0.91%   |
|         Willow         |   45.336    |   0.91%   |
|         crisog         |   43.758    |   0.88%   |
|         Marcie         |   42.800    |   0.86%   |
|         Felix          |   37.883    |   0.76%   |
|       EbonyBelle       |   37.050    |   0.74%   |
|        cortanav        |   33.074    |   0.66%   |
|      coopahtroopa      |   29.845    |   0.60%   |
|         Kokomo         |   26.153    |   0.52%   |
|       ALOYSIOUS        |   21.385    |   0.43%   |
|          peth          |   19.079    |   0.38%   |
|    youngkidwarrior     |   16.243    |   0.32%   |
|          Zizi          |   15.908    |   0.32%   |
|      TheCryptoCPA      |   13.939    |   0.28%   |
|         alive          |   13.275    |   0.27%   |
|         Threm          |   12.538    |   0.25%   |
|        prose11         |   10.477    |   0.21%   |
|        mozilla         |    9.900    |   0.20%   |
|          kus           |    9.720    |   0.19%   |
|        yetifree        |    3.292    |   0.07%   |
|      OttoZastrow       |    2.028    |   0.04%   |
|   Vanilladelphia-eth   |    1.259    |   0.03%   |
|         Dazuck         |    1.065    |   0.02%   |
|      monstrosity       |    0.872    |   0.02%   |
|       metaverde        |    0.790    |   0.02%   |
|         diego          |    0.656    |   0.01%   |
|       johnny1093       |    0.426    |   0.01%   |
|       stargarden       |    0.314    |   0.01%   |
|      decentralion      |    0.000    |   0.00%   |
|         beanow         |    0.000    |   0.00%   |
|      brianlitwin       |    0.000    |   0.00%   |
|        mzargham        |    0.000    |   0.00%   |
|        zixuanzh        |    0.000    |   0.00%   |
|        wchargin        |    0.000    |   0.00%   |
|         amico          |    0.000    |   0.00%   |
|          evan          |    0.000    |   0.00%   |
|        burrrata        |    0.000    |   0.00%   |
|         yalor          |    0.000    |   0.00%   |
|          rpp           |    0.000    |   0.00%   |
|        pet3rpan        |    0.000    |   0.00%   |
|       sourcecred       |    0.000    |   0.00%   |
|        protocol        |    0.000    |   0.00%   |
|         noman          |    0.000    |   0.00%   |
|       deltafreq        |    0.000    |   0.00%   |
|     doctorrobinson     |    0.000    |   0.00%   |
|         jbenet         |    0.000    |   0.00%   |
|     flyingzumwalt      |    0.000    |   0.00%   |
|         sriche         |    0.000    |   0.00%   |
|       ecwireless       |    0.000    |   0.00%   |
|         EthMan         |    0.000    |   0.00%   |
|        jimpick         |    0.000    |   0.00%   |
|        bpierre         |    0.000    |   0.00%   |
|        santigs         |    0.000    |   0.00%   |
|       Jzjallday        |    0.000    |   0.00%   |
|         autark         |    0.000    |   0.00%   |
|       Most-Evil        |    0.000    |   0.00%   |
|        Carla78         |    0.000    |   0.00%   |
|        Macgyver        |    0.000    |   0.00%   |
|        marcvi1a        |    0.000    |   0.00%   |
|      beltenebros       |    0.000    |   0.00%   |
|        cryptouf        |    0.000    |   0.00%   |
|        gilgahex        |    0.000    |   0.00%   |
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200032326400385/1200066164537246) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

File: src/cli/help.js [hunk 1]
```diff
@@ -55,7 +55,6 @@ function usage(print: (string) => void): void {
       AUXILIARY
       load          load plugin data into cache
       graph         build Cred graph from cached plugin data
-      score         generate Cred Scores from from graph
```
Comment (topocount) [inline src/cli/help.js:58]: let's just update this to be `credrank` instead of `score`.  The description is still accurate.

**Review by topocount — COMMENTED** [review]
These are my comments from reviewing the code changes. Will approve after I test locally. 

Forgive me if I'm missing something, but if we aren't going to be supporting timeline cred artifacts from the CLI, why not just remove them? Do we want to keep them around for further analysis?

File: src/cli/help.js (hunk 1 above)
Comment (blueridger) [reply src/cli/help.js:58]: https://github.com/sourcecred/sourcecred/pull/2839

File: src/cli/grain.js [hunk 2]
```diff
@@ -3,11 +3,9 @@
 import fs from "fs-extra";
 import {join} from "path";
 import {loadFileWithDefault, loadJson} from "../util/disk";
-import {fromJSON as credResultFromJson} from "../analysis/credResult";
-import {CredView} from "../analysis/credView";
 import {Ledger} from "../core/ledger/ledger";
-import {applyDistributions} from "../core/ledger/applyDistributions";
-import {computeCredAccounts} from "../core/ledger/credAccounts";
+import {applyDistributions2} from "../core/ledger/applyDistributions";
+import {computeCredAccounts2} from "../core/ledger/credAccounts";
```
[the remark this answers is not in the document]
Comment (blueridger) [reply src/cli/grain.js:8]: that's nifty. either way, doesn't affect a whole lot

**Comment by blueridger** [comment]
> Forgive me if I'm missing something, but if we aren't going to be supporting timeline cred artifacts from the CLI, why not just remove them? Do we want to keep them around for further analysis?

We have a BUNCH of stuff to remove after this PR. In the spirit of more atomic commits, I'm planning to do clean up separately. This is also good practice for the sake of revertability.

File: src/cli/grain.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (blueridger) [reply src/cli/grain.js]: made the change

**Review by topocount — APPROVED** [review]
One last nit. Can we remove the "score" calculation from `go`? It'll be so much faster.

File: src/cli/go.js [hunk 3]
```diff
@@ -26,8 +25,7 @@ const goCommand: Command = async (args, std) => {
   const commandSequence = [
     {name: "load", command: load, args: []},
     {name: "graph", command: graph, args: []},
-    {name: "credrank", command: credrank, args: ["--stealth"]},
-    {name: "score", command: score, args: []},
```
Comment (blueridger) [inline src/cli/go.js:30]: @topocount already did :)

**Pull request merged**